lib/datadog/tracing/tracer.rb

```diff
@@ -376,48 +376,27 @@ def start_span(
       )
         trace = _trace || start_trace(continue_from: continue_from)
 
+        options = {
+          events: SpanOperation::Events.new,
+          on_error: on_error,
+          resource: resource,
+          service: service,
+          tags: resolve_tags(tags),
+          type: type,
+        }
+
         if block
           # Ignore start time if a block has been given
-          trace.measure(
-            name,
-            events: build_span_events,
-            on_error: on_error,
-            resource: resource,
-            service: service,
-            tags: resolve_tags(tags),
-            type: type,
-            &block
-          )
+          trace.measure(name, **options, &block)
         else
           # Return the new span
-          span = trace.build_span(
-            name,
-            events: build_span_events,
-            on_error: on_error,
-            resource: resource,
-            service: service,
-            start_time: start_time,
-            tags: resolve_tags(tags),
-            type: type
-          )
-
-          span.start(start_time)
-          span
+          trace.build_span(name, start_time: start_time, **options).tap do |span|
+            span.start(start_time)
+          end
```

Comment:
Using `**kwargs` it's slower than passing all the arguments.

Also, using `tap` is slower than calling the method directly, then returning the original value.

Reply:
.I recommend passing the values by name, similar to how they were before, and not using `tap`.

Reply:
Don't take my word for it, our microbenchmarks should tell us the real numbers: `benchmarks/tracing_trace.rb`